tests/models/version-test.js

```diff
@@ -193,6 +193,38 @@ module('Model | Version', function (hooks) {
         ],
       );
     });
+
+    test('handles newly released versions correctly', async function (assert) {
+      let crate = this.server.create('crate');
+      this.server.create('version', { crate, num: '0.4.0' });
+      this.server.create('version', { crate, num: '0.4.1' });
+
+      let crateRecord = await this.store.findRecord('crate', crate.name);
+      let versions = (await crateRecord.versions).slice();
+
+      assert.deepEqual(
+        versions.map(it => ({ num: it.num, isHighestOfReleaseTrack: it.isHighestOfReleaseTrack })),
+        [
+          { num: '0.4.0', isHighestOfReleaseTrack: false },
+          { num: '0.4.1', isHighestOfReleaseTrack: true },
+        ],
+      );
+
+      this.server.create('version', { crate, num: '0.4.2' });
+      this.server.create('version', { crate, num: '0.4.3', yanked: true });
+      crateRecord = await this.store.findRecord('crate', crate.name, { reload: true });
+      versions = (await crateRecord.versions).slice();
+
+      assert.deepEqual(
+        versions.map(it => ({ num: it.num, isHighestOfReleaseTrack: it.isHighestOfReleaseTrack })),
+        [
+          { num: '0.4.0', isHighestOfReleaseTrack: false },
+          { num: '0.4.1', isHighestOfReleaseTrack: false },
+          { num: '0.4.2', isHighestOfReleaseTrack: true },
+          { num: '0.4.3', isHighestOfReleaseTrack: true },
```

Comment:
```suggestion
          { num: '0.4.2', isHighestOfReleaseTrack: true },
          { num: '0.4.3', isHighestOfReleaseTrack: true },
```

I assume 0.4.2 is highest because 0.4.3 is yanked, but why does 0.4.3 also have `isHighestOfReleaseTrack: true`?

Reply:
did the previous implementation work the same way? I have a feeling that this was unintentional previously... 😂

Reply:
Ha! I implemented it this way solely to pass the previous tests, although I admit feeling a bit strange with it in the past.

If we would also like to ignore yanked crates and only has one `highestOfReleaseTrack` for each `releaseTrack`, we could simplify the implementation to
```js
  @cached get releaseTrackSet() {
    let map = new Map();
    let { versionsObj: versions, versionIdsBySemver } = this;
    for (let id of versionIdsBySemver) {
      let { releaseTrack, isPrerelease, yanked } = versions[id];
      if (releaseTrack && !isPrerelease && !yanked && !map.has(releaseTrack)) {
        map.set(releaseTrack, id);
      }
    }
    return new Set(map.values());
  }
```

Reply:
yeah, I'd say let's go for the simpler implementation 👍